flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -1439,13 +1439,18 @@ genParallelOp(Fortran::lower::AbstractConverter &converter,
         reductionSyms;
     allSymbols.append(privateSyms);
     for (auto [arg, prv] : llvm::zip_equal(allSymbols, region.getArguments())) {
-      converter.bindSymbol(*arg, prv);
+      fir::ExtendedValue hostExV = converter.getSymbolExtendedValue(*arg);
+      converter.bindSymbol(*arg, hlfir::translateToExtendedValue(
+                                     loc, firOpBuilder, hlfir::Entity{prv},
+                                     /*contiguousHint=*/
+                                     std::holds_alternative<fir::ArrayBoxValue>(
```

Comment:
You could also use `Fortran::evaluate::IsSimplyContiguous(*arg, converter.getFoldingContext())`. It is more generic (will cover the character array case for instance) and more expressive, and it removes the need of calling `getSymbolExtendedValue` that could generate useless code.